### DISCUSSION TITLE
Disaster - TrueNAS used the HDDs of one zpool in creating another!

### DISCUSSION THREAD

**Discussion by Shekelme** [openzfs/zfs#12859] [Q&A]
Hi community!
A terrible thing happened yesterday. TrueNAS used the HDDs of one zpool in creating another...
As a result, the zpool that previously owned the **three disks** involved in the new zpool was damaged because it consisted of 2 **raidz2** vdevs.
My mistake - I should have first figured out which three "extra disks" TrueNAS sees in the "disks" under "Storage/Pools/Create Pool",
which physically should not be there. Usually only disks not involved in arrays are displayed here. I trusted TrueNAS, I could not admit that it could dispose of the disks in this way, and three "extra" disks are some kind of glitch.

So my stupid question with an almost guaranteed known answer is: _can I restore the raidz2 vdev (and thus the entire pool) in case of 3 drives failure/absence_? May be there is any magic to "_unformat_" that 3 drives detached from the affected zpool in the process of creating a new zpool? Anything else? Please...

**Comment by amotin** [comment]
I have no magic recipe how to "unfortmat" the drives.  It may require some professional data recovery.  There is a way to import the pool read-only without one of the top-level vdevs, that may allow you to read structure of datasets and files due to additional metadata duplication in ZFS, but access to any data block that was written to the affected top-level vdev will fail, so likely any file of significant size will get some holes.
